### ISSUE TITLE
box-shadow rules with space gets detected as LTR != RTL

### ISSUE THREAD

**Issue by hyperknot** [facebook/stylex#1120]
### Describe the issue

I have a rule like this:

```
boxShadow: '0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)',
```

The key is on the space after the comma.

It generates the following CSS:

```
html:not([dir='rtl']) .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1),0 4px 6px -4px rgb(0 0 0 / .1)}
html[dir='rtl'] .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1), 0 4px 6px -4px rgb(0 0 0 / .1)}
```

Debug lines:

```
  cssLTR: .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1),0 4px 6px -4px rgb(0 0 0 / .1)}
  cssRTL: .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1), 0 4px 6px -4px rgb(0 0 0 / .1)}
  cssRTL === null: false
  cssRTL === cssLTR: false
  priority: 3000
  🚨 RTL differs from LTR!
  LTR length: 94
  RTL length: 95
```

### Expected behavior

Should be detected correctly.

### Steps to reproduce

Add the rule

**Comment by necolas** [comment]
> Should be detected correctly.

Sorry what are you asking? Are you suggesting that boxShadow shouldn't get modified for RTL?

**Comment by hyperknot** [comment]
I'm reporting a bug. The bug is the following:

If this rule contains a space after the comma, like below:
```
boxShadow: '0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)',
```

Then it gets mistakenly recognised as if an RTL version was set, and separate LTR and RTL versions are generated, as shown below:

```
html:not([dir='rtl']) .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1),0 4px 6px -4px rgb(0 0 0 / .1)}
html[dir='rtl'] .boxShadow-xalsa2k{box-shadow:0 10px 15px -3px rgb(0 0 0 / .1), 0 4px 6px -4px rgb(0 0 0 / .1)}
```

The correct behaviour is to not generate two versions, as RTL dir was not specified.

**Comment by necolas** [comment]
I don't think it has anything to do with a comma. In StyleX you don't "set" an RTL version, the compiler creates them in some circumstances.

`boxShadow` has a legacy behavior where RTL versions of shadows are created, see the unit tests here - https://github.com/facebook/stylex/blob/main/packages/%40stylexjs/babel-plugin/__tests__/legacy/transform-logical-values-test.js#L281

It could also be that this behavior has bugs. But StyleX should not create RTL versions of boxShadow, cursor, and other non-logical properties at all. We'll put this behavior behind a feature flag, as we need to migrate away from this behavior internally.

**Comment by hyperknot** [comment]
What makes a boxShadow legacy vs non-legacy?

I tried and removing the space after the comma doesn't matter, for RTL it always inserts that space. I mean I guess that's where the bug lies, RTL != LTR because of that space after the comma.

**Comment by hyperknot** [comment]
OK, from the test I guess the legacy behaviour is to negate the horizontal offset automatically.
But here, nothing is negated, just a space is inserted silently, causing the difference.

**Comment by nmn** [comment]
The `html:not([dir='rtl'])` rules should only ever be generated in the legacy setting.

Further, box-shadows should not be flipped for RTL at all. This was originally implemented to match what Meta used previously, but after some research, I came to the conclusion that the majority of users that use RTL languages don't expect two elements to flip horizontally:
1. scrollbars. Most RTL users still expect them on the right
2. Shadows. They expect the "light source" the be in the top-left just like LTR users.
